src/utils/span-pointers.ts

```diff
@@ -0,0 +1,82 @@
+import { eventTypes } from "../trace/trigger";
+import { logDebug } from "./log";
+
+interface SpanPointerAttributes {
+  pointerKind: string;
+  pointerDirection: string;
+  pointerHash: string;
+}
+
+/**
+ * Computes span pointer attributes
+ *
+ * @param {eventTypes} eventSource - The type of event being processed (e.g., S3, DynamoDB).
+ * @param {any} event - The event object containing source-specific data.
+ * @returns {SpanPointerAttributes[] | undefined} An array of span pointer attribute objects, or undefined if none could be computed.
+ */
+export function getSpanPointerAttributes(
+  eventSource: eventTypes | undefined,
+  event: any,
+): SpanPointerAttributes[] | undefined {
+  if (!eventSource) {
+    return;
+  }
+
+  switch (eventSource) {
+    case eventTypes.s3:
+      return processS3Event(event);
+  }
+}
+
+function processS3Event(event: any): SpanPointerAttributes[] {
+  const records = event.Records || [];
+  const spanPointerAttributesList: SpanPointerAttributes[] = [];
+
+  // Get dependencies from tracer only when needed
+  let constants;
+  let util;
+  try {
+    constants = require("dd-trace/packages/dd-trace/src/constants");
+    util = require("dd-trace/packages/dd-trace/src/util");
```

Comment:
@duncanista, can you help answer a question for me?

I recall when you added support for w3c trace propagation to datadog-lambda-js, you said that instead of directly importing the tracer from dd-trace, we create an interface to wrap it. 

If that's correct, then is there any issue with us requiring other files from ddtrace here?

Reply:
There might be an issue when customer is not using the tracer (some do it), it would essentially crash, ideally, we'd have a better way to get this type of data

Reply:
Could you take another look at the updated code from my latest commit? It should be safe and avoid any crashing:
```typescript
  let S3_PTR_KIND;
  let SPAN_POINTER_DIRECTION;
  let generatePointerHash;
  try {
    const constants = require("dd-trace/packages/dd-trace/src/constants");
    const util = require("dd-trace/packages/dd-trace/src/util");

    ({ S3_PTR_KIND, SPAN_POINTER_DIRECTION } = constants);
    ({ generatePointerHash } = util);
  } catch (err) {
    if (err instanceof Error) {
      logDebug("Failed to load dd-trace span pointer dependencies", err);
    }
    return spanPointerAttributesList;
  }
```

Reply:
@duncanista, Can you give more details (or point me to a doc somewhere) about how "customer is not using the tracer" is implemented? Do you mean they just aren't including `ddtrace` in their package.json file? I'd like to be able to create a small sample app that I can test this PR with that "is not using the tracer".

Reply:
Yeah correct, just the lambda library, not the tracer. Let me review the whole PR